File: Linked List/Intro.java
class Node {
  int data;
  Node next;


  Node(int data) {
    this.data = data;
    this.next = null;
  }

  Node(int data, Node next) {
    this.data  = data;
    this.next = next;
  }
}

public class Main {
  public static void main(String[] args) {
    Node node = new Node(20);
    Node  node2 = new Node(10);
    Node node3 = new Node(30);

    node2.next = node;
    node.next = node3;
    display(node2);
  }

  static void display(Node head) {
    Node temp = head;
    while(temp != null) {
      System.out.print(temp.data + "->");
      temp = temp.next;
    }
    System.out.println("END");
  }
}

File: Linked List/SLL/intro.cpp
#include <iostream>
#include<bits/stdc++.h>
using namespace std;

class Node {
public:
    int data;
    Node* next;

  Node() {
    this->next = NULL;
  }

  Node(int data) {
    this->data = data;
    this->next = NULL;
    }
};

void printLL(Node* head) {
  int cnt = 0;
  Node* temp = head;
  while(temp != NULL) {
    cout << temp->data << "->";
    temp = temp->next;
    cnt++;
  }
  cout <<"END" <<endl; 
  cout <<cnt <<endl;
}

int main() {
  Node* first = new Node(10);
  Node* second = new Node(20);
  Node* third = new Node(30);
  first->next = second;
  second->next = third;
  Node* head = first;
  printLL(head);
}